unstract/connectors/src/unstract/connectors/filesystems/google_drive/google_drive.py

```diff
@@ -25,13 +27,16 @@ class GoogleDriveFS(UnstractFileSystem):
     # }
     def __init__(self, settings: dict[str, Any]):
         super().__init__("GoogleDrive")
-        client_secrets = json.loads(
-            GCSHelper().get_secret("google_drive_client_secret")
-        )
+        try:
+            self.client_secrets = json.loads(
+                GCSHelper().get_secret("google_drive_client_secret")
+            )
+        except GoogleApiException.PermissionDenied as e:
+            raise GoogleDriveAccessDeniedException(detail=e.details) from e
```

Comment:
@kirtimanmishrazipstack we'd throw this message for all such `PermissionDenied` errors. But the error message is too specific. Either catch and raise only for the specific scenario or have a generic message that encompasses all this.
Also a cloud user might not find such a message useful - he wouldn't know what env we're talking about here

Reply:
@chandrasekharan-zipstack as discussed, handled over generic message